mediapipe/graphs/instantmotiontracking/calculators/matrices_manager_calculator.cc

```diff
@@ -0,0 +1,324 @@
+        // with the direction of finger movement from the user (system dependent)
+        Eigen::AngleAxisf(-rotation_radians, Eigen::Vector3f::UnitY()) *
+        Eigen::AngleAxisf(0, Eigen::Vector3f::UnitZ()) *
+        Eigen::AngleAxisf(0, Eigen::Vector3f::UnitX());
+  // Matrix must be transposed due to the method of submatrix generation in Eigen
+  return user_rotation_submatrix.transpose();
+}
+
+// Using a specified scale factor, generate a scaling matrix for use with base rotation submatrix
+const DiagonalMatrix3f MatricesManagerCalculator::GenerateScalingMatrix(const float scale_factor) {
+  DiagonalMatrix3f scaling_matrix(scale_factor, scale_factor, scale_factor);
+  return scaling_matrix;
+}
+
+// TODO: Investigate possible differences in warping of tracking speed across screen
+// Using the sticker anchor data, a translation vector can be generated in OpenGL coordinate space
+const Vector3f MatricesManagerCalculator::GenerateAnchorVector(const Anchor tracked_anchor) {
+  // Using an initial_z value in OpenGL space, generate a new base z-axis value to mimic scaling by distance.
+  const float z = initial_z_ * tracked_anchor.z;
+
+  // Using triangle geometry, the minimum for a y-coordinate that will appear in the view field
+  // for the given z value above can be found.
+  const float y_minimum = z * (tan(vertical_fov_radians_ / 2));
+
+  // The aspect ratio of the device and y_minimum calculated above can be used to find the
+  // minimum value for x that will appear in the view field of the device screen.
+  const float x_minimum = y_minimum * (1.0 / aspect_ratio_);
+
+  // Given the minimum bounds of the screen in OpenGL space, the tracked anchor coordinates
+  // can be converted to OpenGL coordinate space.
+  //
+  // (i.e: X and Y will be converted from [0.0-1.0] space to [x_minimum, -x_minimum] space
+  // and [y_minimum, -y_minimum] space respectively)
+  const float x = (-2 * tracked_anchor.x * x_minimum) + x_minimum;
+  const float y = (-2 * tracked_anchor.y * y_minimum) + y_minimum;
+
+  // A translation transformation vector can be generated via Eigen
+  const Vector3f t_vector(x,y,z);
+  return t_vector;
+}
+
+// Using the yaw, pitch, and roll, a rotation submatrix can be generated,
+// universal to each object appearing in the device view
+const Matrix3f MatricesManagerCalculator::GenerateIMURotationSubmatrix(
+  const float yaw, const float pitch, const float roll) {
+  Eigen::Matrix3f r_submatrix;
+    r_submatrix =
+        // The yaw value is associated with the Y-axis.
+        Eigen::AngleAxisf(yaw, Eigen::Vector3f::UnitY()) *
+        // The roll value is associated with the Z-axis.
+        Eigen::AngleAxisf(roll, Eigen::Vector3f::UnitZ()) *
+        // The pitch value is associated with the X-axis.
+        // The (-M_PI/2) must be added in order to adjust the default
+        // rendering of the object (the object should appear in the upright
```

Comment:
Note: This should be handled like "default scaling".  It sounds like at the moment all the obj's have the same default orientation, so this is fine for now.